pkg/module/validation.go

```diff
@@ -3,11 +3,150 @@ package module
 import (
 	"errors"
 	"fmt"
+	"os"
+	"path/filepath"
 	"strings"
+	"time"
 
+	"github.com/kyma-project/cli/internal/kube"
+	"go.uber.org/zap"
+	"gopkg.in/yaml.v3"
 	amv "k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
 )
 
+const defaultCRName = "default.yaml"
+
+//readDefaultCR reads the default CR file's contents. The first returned value is true if the file exists. It's false if the file does not exist or an error occurred.
+func readDefaultCR(modulePath string) (bool, []byte, error) {
+	//TODO: Do we need to override the name or it's always "default.yaml"?
+	crPath := filepath.Join(modulePath, defaultCRName)
+
+	fileInfo, err := os.Stat(crPath)
+	if errors.Is(err, os.ErrNotExist) {
+		return false, nil, nil
+	}
+	if err != nil {
+		return false, nil, fmt.Errorf("Error accessing the default CR file `%q`: %w", crPath, err)
+	}
+
+	if !fileInfo.Mode().IsRegular() {
+		return false, nil, fmt.Errorf("Error reading the default CR file `%q`: Not a regular file", crPath)
+	}
+
+	crData, err := os.ReadFile(crPath)
+	if err != nil {
+		return false, nil, fmt.Errorf("Error reading the default CR file `%q`: %w", crPath, err)
+	}
+
+	return true, crData, nil
+}
+
+//ensureDefaultNamespace parses crData and ensures that metadata.namespace is set to the value "default". This is because of how we use the envtest to validate the CR.
```

Comment:
```suggestion
//ensureDefaultNamespace parses crData and ensures that the metadata.namespace attibute exists and it's value is "default". This is because of how we use the envtest to validate the CR.
```